### PR TITLE
Add get shared_secret & root_entropy FFIs to LibMobileCoin

### PR DESCRIPTION
**Pull request by the-real-adammork**
Add getters for shared_secret and root_entropy to LibMobileCoin

### Motivation

Getters are needed for shared_secret, and root_entropy requirements have changed so the removed implementation has been re-added.

[Loose Ends - Lets Wax a Fatty](https://www.youtube.com/watch?v=CBN5c1E8nZI)

### CODE REVIEW

**Review by voloshyn — APPROVED** [review]

**Review by jcape — CHANGES_REQUESTED** [review]
Let's hold off on this until v1.2.1.

**Comment by the-real-adammork** [comment]
@jcape ok I need one more review

**Review by jcape — APPROVED** [review]

**Pull request merged**
